.github/workflows/puppetfile.yml

```
@@ -0,0 +1,73 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Puppetfile checks
+
+# Controls when the workflow will run
+on:
+  workflow_call:
+    inputs:
+      container_image:
+        description: Image to use when running tests
+        default: 'puppet/puppet-dev-tools:2022-03-28-92c7176'
+        required: false
+        type: string
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  r10k_validate:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container_image }}
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+#      - name: github known host
+#        run: mkdir /~/.ssh/ && echo githubPubKey >> /~/.ssh/known_hosts
+      - name: validate Puppetfile
+        run: rake -f /Rakefile r10k:validate
+  r10k_check_dups:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container_image }}
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Check for puppetfile duplicates
+        run: rake -f /Rakefile r10k:duplicates
+  r10k_syntax:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container_image }}
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Check for puppetfile syntax correctness
+        run: rake -f /Rakefile r10k:syntax
+  r10k_deprecation:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container_image }}
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Check for puppetfile module deprecations
+        run: rake -f /Rakefile r10k:deprecation          
+  r10k_solve_dependencies:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container_image }}
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      - run: mkdir -p /github/home/.cache
+        name: "Create cache directory"
```

That this is needed feels like a bug in the tooling. Also, wouldn't this work?
```suggestion
      - run: mkdir -p $HOME/.cache
        name: "Create cache directory"
```

---

Correct, I think this job should be removed until the tooling can be fixed as there were thrown exceptions as well.   Although maybe it is just my puppetfile